src/utils/fetch-transport.ts

```diff
@@ -1,10 +1,17 @@
-import fetch, { Headers } from 'node-fetch';
-import { HttpRequest } from './http-request';
+import { Transport } from './transport';
 
-export class HttpRequestFetch extends HttpRequest {
-  // constructor() { }
+export class FetchTransport implements Transport {
+  private fetch: (...params: any[]) => Promise<Response>;
 
-  public post<ResponseType>(url: string, content: any, headers: { [headerName: string]: string }) {
+  constructor(fetch: any) {
+    this.fetch = fetch;
```

Comment:
This seems sketchy. You already defined a type for fetch `(...params: any[]) => Promise<Response>`